### ISSUE TITLE
rust-analyzer doesn't see a problem in `compile_error!("")`

### ISSUE THREAD

**Issue by MedzikUser** [rust-lang/rust-analyzer#12736]
If you use `compile_error!("")` then rust-analyzer does not show the error, and when you use, for example, `compile_error!("a")` it already sees the

Example:

1. rust-analyzer doesn't see a problem
```rs
fn test() {
    compile_error!("")
}
```

2. rust-analyzer see a problem
```rs
fn test() {
    compile_error!("a")
}
```

cargo check detect problem

![image](https://user-images.githubusercontent.com/87065584/178146204-84310b45-4de4-4a62-b321-2cdcebb084d1.png)

Example in IDE:

![image](https://user-images.githubusercontent.com/87065584/178146322-61d724fb-be6a-4e61-ac6e-064fb007528f.png)

*I know it all makes no sense

**rust-analyzer version**: rust-analyzer 0.0.0 (75b22326d 2022-07-03)

**rustc version**: rustc 1.62.0 (a8314ef7d 2022-06-27)

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTUP_HOME` or `CARGO_HOME`)

**Comment by lowr** [comment]
Duplicate of #11404; VSCode treats empty error messages specially and filters them out.
